### ISSUE TITLE
Need a way to disable path parse for gen csv

### ISSUE THREAD

**Issue by aharbis** [operator-framework/operator-sdk#2983]
## Feature Request

**Is your feature request related to a problem? Please describe.**

Yes. The behavior is summarized in the [doc](https://sdk.operatorframework.io/docs/golang/references/markers/):

>The SDK also checks path elements against a list of well-known path to x-descriptor string mappings and either uses a match as x-descriptors, or does not set x-descriptors.

**Describe the solution you'd like**

However, if I have a path such as `resources.requests.cpu` (because we are wanting to manually define how resources are provided), and I declare this as a `number` through the `specDescriptors`, the SDK automatically adds the full `urn:alm:descriptor:com.tectonic.ui:resourceRequirements` descriptor on top of mine.

```go
// Custom CPU
// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="CPU"
// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:number,urn:alm:descriptor:com.tectonic.ui:fieldGroup:Custom Resources"
CPU int `json:"cpu,omitempty"`
```

```yaml
- description: Custom CPU
  displayName: CPU
  path: resources.requests.cpu
  x-descriptors:
  - urn:alm:descriptor:com.tectonic.ui:fieldGroup:Custom Resources
  - urn:alm:descriptor:com.tectonic.ui:number
  - urn:alm:descriptor:com.tectonic.ui:resourceRequirements
```

Notice that the `x-descriptors` output has my `fieldGroup` and `number` defined, but then on top of that there is `resourceRequirements`. This means that in the OpenShift Console UI `Edit Form` view, the entire [resourceRequirements](https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#2-resourcerequirements) component is rendered where I would expect a single box for a number.

I would like to be able to tell the SDK to **not** automatically parse the `path` or **not** make decisions about type without my telling it to do so.

**Comment by estroz** [comment]
@aharbis we could skip auto-selecting `x-descriptors` if an `x-descriptors` annotation is present on a field.

Note that with #2792, when you migrate to the new marker format, you get the behavior for free because the SDK will no longer auto-select `x-descriptors`. Obviously that's not a fix because it isn't a released feature yet; we should implement my suggestion or something similar for the current annotation format.

/kind feature
/good-first-issue
/help-wanted
/milestone v0.18.0

**Comment by pratikjagrut** [comment]
> Note that with #2792, when you migrate to the new marker format, you get the behavior for free because the SDK will no longer auto-select `x-descriptors`. Obviously that's not a fix because it isn't a released feature yet; we should implement my suggestion or something similar for the current annotation format.

Hi, @estroz I did not understand if this issue is still open for PRs because you said #2792 would by default solve this issue. 
If this issue is open for PR I wanna contribute to it as this is marked as a good first issue, if no one is working on it.

**Comment by estroz** [comment]
@pratikjagrut I've removed those labels since the old annotation format will be removed by v1.0 and this issue is fixed by the new marker format.

**Comment by estroz** [comment]
Closed by #2792
